isis/src/clipper/objs/ClipperPushBroomCamera/ClipperPushBroomCamera.cpp

```diff
@@ -0,0 +1,129 @@
+/** This is free and unencumbered software released into the public domain.
+
+The authors of ISIS do not claim copyright on the contents of this file.
+For more details about the LICENSE terms and the AUTHORS, you will
+find files of those names at the top level of this repository. **/
+
+/* SPDX-License-Identifier: CC0-1.0 */
+
+#include "ClipperPushBroomCamera.h"
+
+#include "CameraDistortionMap.h"
+#include "CameraFocalPlaneMap.h"
+#include "iTime.h"
+#include "LineScanCameraGroundMap.h"
+#include "LineScanCameraSkyMap.h"
+#include "NaifStatus.h"
+#include "LineScanCameraDetectorMap.h"
+
+namespace Isis {
+  /**
+   * Constructs a ClipperPushBroomCamera object using the image labels.
+   *
+   * @param Cube &cube Clipper EIS image.
+   */
+   ClipperPushBroomCamera::ClipperPushBroomCamera(Cube &cube) : LineScanCamera(cube) {
+
+     m_spacecraftNameLong = "Europa Clipper";
+     m_spacecraftNameShort = "Clipper";
+
+     int frameCode = naifIkCode();
+
+     // ik values for NAC: -159101(general) and filters between -159121 and -159131
+     if (frameCode == -159101 || (frameCode > -159132 && frameCode < -159120)) {
+       m_instrumentNameLong  = "Europa Imaging System Push Broom Narrow Angle Camera";
+       m_instrumentNameShort = "EIS-PBNAC";
+     }
+     // ik values for WAC: -159102(general) and filters between -159141 and -159151
+     else if (frameCode == -159102 || (frameCode > -159152 && frameCode < -159140)) {
+       m_instrumentNameLong  = "Europa Imaging System Push Broom Wide Angle Camera";
+       m_instrumentNameShort = "EIS-PBWAC";
+     }
```

Comment:
This looks solid! Only other potential codes are:
```
      EUROPAM_EIS_NAC_RAD        -159110
      EUROPAM_EIS_WAC_RAD        -159111

      EUROPAM_EIS_NAC_KOZ        -159112
      EUROPAM_EIS_WAC_KOZ        -159113
```

Reply:
I talked to Stuart about these and he was unsure what these codes will be used for. I can go ahead and include them if you believe they are push broom codes.

Reply:
Added in the codes above.